src/nupic/types/Sdr.hpp

```diff
@@ -124,14 +123,25 @@ typedef function<void()>      SDR_callback_t;
  */
 class SparseDistributedRepresentation : public Serializable
 {
-protected:
+private:
```

Comment:
Please can these be protected?

Reply:
yep. does Proxy need it?

Reply:
Sorry. I looked into this more and I think you were right, that it's probably better for these to be private.  I just added a commit which re-does this.

I also made the callbacks private since they have setter methods which must be used.